api/service/stagedstreamsync/staged_stream_sync.go

```diff
@@ -642,8 +642,9 @@ func (ss *StagedStreamSync) addConsensusLastMile(bc core.BlockChain, cs *consens
 			case errors.Is(err, core.ErrNotLastBlockInEpoch):
 			case err != nil:
 				return errors.Wrap(err, "failed to InsertChain")
+			default:
```

Comment:
This line prevents the following code from executing, so no new hashes added to slice. Later we check count of hashes, and if it's not zero, we start sync again and again, which cause infinity loop.

Reply:
but if `err != nil` is true then we'll also never hit the `default` case

Reply:
is this links to the `Block exists` issue ?

Reply:
if `err != nil` we don't need default case and should return error.

Reply:
@sophoah current code considers existed blocks and counts them as new added blocks. adding default helps to not count them as new added blocks because the block hash won't be added to slice. So, the sync loop checks number of hashes and it is zero, so it breaks.